platform-api/che-core-api-core/src/main/java/org/eclipse/che/api/core/rest/HttpJsonRequest.java

```diff
@@ -0,0 +1,409 @@
+                                                                         ServerException,
+                                                                         UnauthorizedException,
+                                                                         ForbiddenException,
+                                                                         NotFoundException,
+                                                                         ConflictException,
+                                                                         BadRequestException;
+
+    /**
+     * Makes http request with content type "application/json" and authorization headers
+     * based on current {@link EnvironmentContext#getCurrent() context}.
+     *
+     * <p>Example:
+     * <pre>{@code
+     *      Set<UsersWorkspaceDto> workspaces = requestFactory.fromUri(apiEndpoint + "/workspace/config")
+     *                                                        .useGetMethod()
+     *                                                        .request(new TypeToken<Set<UsersWorkspace>>() {}.getType());
+     * }</pre>
+     *
+     * @param <T>
+     *         type of the response
+     * @param clazz
+     *         class of the response
+     * @return response parsed to a type
+     * @throws IOException
+     *         when server response content type is different from "application/json"(Not acceptable)
+     * @throws IOException
+     *         when any io error occurs
+     * @throws ServerException
+     *         when response code is 500 or it is different from 400, 401, 403, 404, 409
+     * @throws ForbiddenException
+     *         when response code is 403
+     * @throws NotFoundException
+     *         when response code is 404
+     * @throws UnauthorizedException
+     *         when response code is 401
+     * @throws ConflictException
+     *         when response code is 409
+     * @throws BadRequestException
+     *         when response code is 400
+     * @throws NullPointerException
+     *         when {@code typeOfT} is null
+     * @throws NullPointerException
+     *         when {@code clazz} is null
+     */
+    <T> T request(@NotNull Class<T> clazz, @Nullable Type typeOfT) throws IOException,
+                                                                          ServerException,
+                                                                          UnauthorizedException,
+                                                                          ForbiddenException,
+                                                                          NotFoundException,
+                                                                          ConflictException,
+                                                                          BadRequestException;
+
+    /**
+     * Uses {@link HttpMethod#GET} as a request method.
+     *
+     * @return this request instance
+     */
+    default HttpJsonRequest useGetMethod() {
```

Comment:
JAX-RS 2.0 api is using simple .get(), put(entity), post(entity), delete() and head()